### PR TITLE
Libreactor Refactor

### PR DESCRIPTION
**Pull request by talawahtech**
Code refactoring and simplification. No performance impacting changes.

### CODE REVIEW

File: frameworks/C/libreactor/libreactor.dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
-FROM ubuntu:18.04 as builder
+FROM ubuntu:20.04 as builder
 
 RUN apt-get update -yqq
-RUN apt-get install -yqq wget make automake libtool file gcc-8 g++-8
+RUN apt-get install -yqq wget make automake libtool file gcc-9 g++-9
```
Comment (volyrique) [inline frameworks/C/libreactor/libreactor.dockerfile:4]: GCC 9 is the default version in Ubuntu 20.04, so there is no need to specify it. In fact, you really shouldn't hardcode the version (keeping the dependency versions fixed is served by hardcoding the base image version) unless you have a very good reason to. Just think about the moment when Ubuntu 22.04 comes out - ideally that would be a one line change, but in your case it won't.

Comment (talawahtech) [reply frameworks/C/libreactor/libreactor.dockerfile:4]: @volyrique @nbrady-techempower I wanted the dependency to be explicit. I don't want the gcc version to be automatically upgraded. For example gcc-10 has a backwards incompatible change that I know will cause this code to no longer build.

Comment (volyrique) [reply frameworks/C/libreactor/libreactor.dockerfile:4]: It won't be upgraded as long as you keep the base image the same.

Comment (talawahtech) [reply frameworks/C/libreactor/libreactor.dockerfile:4]: @volyrique right, but I don't want the gcc version to be updated automatically when I update the base image. I am deliberately pinning the dependency on that specific version of gcc.

**Pull request merged**